src/sentry/auth/access.py

```diff
@@ -325,13 +330,15 @@ def __init__(
         )
 
     def has_team_access(self, team: Team) -> bool:
+        assert self._member is not None
```

Comment:
It's impossible to hit provided the types on `OrganizationMemberAccess.__init__` are respected -- that is, `member` on this particular subclass is `OrganizationMember`, not `OrganizationMember | None`.

If `self._member` were `None`, it would error out anyway, on all cases except `team.status != TeamStatus.VISIBLE`.